vendor/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_image.go

```diff
@@ -117,7 +122,7 @@ func (o *ImageOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []st
 	o.ChangeCause = f.Command(cmd, false)
 	o.DryRun = cmdutil.GetDryRunFlag(cmd)
 	o.Output = cmdutil.GetFlagString(cmd, "output")
-	o.ResolveImage = f.ResolveImage
+	o.ResolveImage = resolveImageFactory(f, cmd)
```

Comment:
can we get rid of this field for set image and just call the function directly in Run() ? Also remove it from upstream factory as it is not used anywhere else.